### DISCUSSION TITLE
What's the quickest way to extract permissions and intentes using AndroGuard?

### DISCUSSION THREAD

**Discussion by ddxv** [androguard/androguard#998] [Q&A]
I tried:

```py
from androguard.misc import AnalyzeAPK

# Load the APK
a, d, dx = AnalyzeAPK(apk)

# Extract and Print AndroidManifest.xml
android_manifest = a.get_android_manifest_xml()
print(android_manifest)

# List Permissions
permissions = a.get_permissions()
print("Permissions:")
for permission in permissions:
    print(permission)

 ```

But noticed that the extract step seemed to take a very long time. Is there another way that skips unpacking all the resources?

**Comment by erev0s** [comment]
@ddxv  you might want to try using `apk.APK('apk/path/here')`
the `AnalyzeAPK` includes the step to get the apk stuff (its the `a` you have there) but for the permissions etc you dont need it.
